migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs

```diff
@@ -173,3 +258,23 @@ impl SqlFlavour for MysqlFlavour {
         Ok(sql_schema)
     }
 }
+
+#[derive(BitFlags, Debug, Clone, Copy, PartialEq)]
+#[repr(u8)]
+pub enum Circumstances {
+    LowerCasesTableNames = 0b0001,
+    IsMysql56 = 0b0010,
+    IsMariadb = 0b0100,
+}
+
+fn check_datamodel_for_mysql_5_6(datamodel: &Datamodel, errors: &mut Vec<String>) {
```

Comment:
Should this kind of things be in the data model validator?

Reply:
It can't be because we don't know the database version in the schema parser :/